jdaviz/core/template_mixin.py

```diff
@@ -230,32 +302,50 @@ def _update_has_subregions(self):
     def labels(self):
         return [s['label'] for s in self.items if 'label' in s.keys()]
 
+    @cached_property
+    def selected_item(self):
+        for item in self.items:
+            if item['label'] == self.selected:
+                return item
+        return {}
+
     @cached_property
     def selected_obj(self):
-        if self.selected == "Entire Spectrum":
+        if self.selected == self._default_text:
             return None
-        return self.app.get_subsets_from_viewer("spectrum-viewer",
-                                                subset_type="spectral").get(self.selected)
+        subset_type = self.selected_item['type']
+        # NOTE: we use reference names here instead of IDs since get_subsets_from_viewer requires
+        # that.  For imviz, this will mean we won't be able to loop through each of the viewers,
+        # but the original viewer should have access to all the subsets.
+        for viewer_ref in self.viewer_refs:
+            match = self.app.get_subsets_from_viewer(viewer_ref,
+                                                     subset_type=subset_type).get(self.selected)
+            if match is not None:
+                return match
```

Comment:
If there isn't a match made (not sure why this would happen, but who knows if a state mismatch happens at some point), no action would be taken, right? Shouldn't we through an error/exception?

Reply:
Looks like it just `return None` if there is no match.

Reply:
Not sure if it is more correct to crash or return None if there is no match though, I'll defer to @kecnry .

Reply:
right now I think returning None from something this low-level is ok.  Any higher-level code can then check against None if it needs to instead of having to wrap in a try-except.  But realistically it should never happen since the choices are synced against the subsets and manually entering an invalid string raises an error in the `_selected_changed` method.